src/ResourceManager/Compute/Commands.Compute/Common/Utils.cs

```diff
@@ -0,0 +1,151 @@
+
+                      "SQLGLCore", "Standard_D4_v3", "Standard_D4s_v3", "Standard_D2_v2", "Standard_DS2_v2",
+
+                      "Standard_E4_v3", "Standard_E4s_v3", "Standard_F2", "Standard_F2s", "Standard_F4s_v2",
+
+                      "Standard_D11_v2", "Standard_DS11_v2", "AZAP_Performance_ComputeV17C"};
+
+        static readonly string[] FourCoreSizesForAccelNet = { "Standard_D3_v2", "Standard_D3_v2_Promo", "Standard_D3_v2_ABC", "Standard_DS3_v2",
+
+                           "Standard_DS3_v2_Promo", "Standard_D12_v2", "Standard_D12_v2_Promo", "Standard_D12_v2_ABC",
+
+                           "Standard_DS12_v2", "Standard_DS12_v2_Promo", "Standard_F8s_v2", "Standard_F4",
+
+                           "Standard_F4_ABC", "Standard_F4s", "Standard_E8_v3", "Standard_E8s_v3", "Standard_D8_v3",
+
+                           "Standard_D8s_v3"};
+
+        sealed class ImageTypeForAccelNet
+        {
+            public string Publisher { get; }
+            public string Offer { get; }
+            public string SkuPattern { get; }
+
+            public ImageTypeForAccelNet(string publisher, string offer, string skuPattern)
+            {
+                Publisher = publisher;
+                Offer = offer;
+                SkuPattern = skuPattern;
+            }
+
+            public bool Matches(string publisher, string offer, string sku)
+            {
+                return (Publisher.Equals(publisher) && (offer == null || offer.Equals(Offer)) && (sku == null || SkuPattern == null || (new Regex(SkuPattern)).IsMatch(sku)));
+            }
+        }
+
+        static readonly ImageTypeForAccelNet[]DistributionsForAccelNet = { new ImageTypeForAccelNet("canonical", "UbuntuServer", "^16.04"), new ImageTypeForAccelNet("suse", "sles", "^12-sp3"),
+                   new ImageTypeForAccelNet("redhat", "rhel", "^7.4"), new ImageTypeForAccelNet("openlogic", "centos", "^7.4"), new ImageTypeForAccelNet("coreos", "coreos", null),
+                   new ImageTypeForAccelNet("credativ", "debian", "-backports"), new ImageTypeForAccelNet("oracle", "oracle-linux", "^7.4"),
+                   new ImageTypeForAccelNet("MicrosoftWindowsServer", "WindowsServer", "^2016"), new ImageTypeForAccelNet("MicrosoftWindowsServer", "WindowsServer", "^2012-R2") };
+
+        public static bool DoesConfigSupportAcceleratedNetwork(Client client, ImageAndOsType imageInfo, string size, string location, string defaultLocation)
+        {
+            if (imageInfo?.Image?.Publisher == null)
+            {
+                return false;
+            }
+
+            var sizeFound = ValidSizesForAccelNet.Where(x => x.ToLower().Equals(size.ToLower()));
+            if (sizeFound == null || sizeFound.Count() <= 0)
+            {
+                return false;
+            }
+
+            var fourCoreSize = FourCoreSizesForAccelNet.Where(x => x.ToLower().Equals(size.ToLower()));
+
+            if (fourCoreSize == null | fourCoreSize.Count() == 0)
```

Comment:
@praries880 should the operator here be `||` ?

Reply:
:) why.. yes.. yes it does